workspace/src/common/launch/art_launch/launch/art.launch.py

```diff
@@ -0,0 +1,98 @@
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.#
+
+# ros imports
+from launch import LaunchDescription
+from launch.actions import SetLaunchConfiguration
+from launch.substitutions import PythonExpression
+from launch.conditions import IfCondition
+from launch_ros.actions import ComposableNodeContainer
+
+# internal imports
+from launch_utils import IncludeLaunchDescriptionWithCondition, GetLaunchArgument, AddLaunchArgument, SetLaunchArgument
+
+def generate_launch_description():
+    ld = LaunchDescription()
+
+    # ----------------
+    # Launch Arguments
+    # ----------------
+
+    AddLaunchArgument(ld, "use_sim", "False")
+    AddLaunchArgument(ld, "use_sim_time", "False")
+    SetLaunchArgument(ld, "use_sim_time", "True", condition=IfCondition(GetLaunchArgument("use_sim")))
+   
+    AddLaunchArgument(ld, "container", "") 
+    container_name = AddLaunchArgument(ld, "container_name", "art_container") 
+
+    # -------------
+    # Composability
+    # -------------
+
+    # If composability is desired, all included launch descriptions should attach to this container and use intraprocess communication
+
+    use_composability = IfCondition(AddLaunchArgument(ld, "use_composability", "False"))
+
+    # If a container name is not provided,
+    # set the name of the container launched above for image_proc nodes
+    set_container_name = SetLaunchConfiguration(
+        condition=use_composability,
+        name='container',
+        value=container_name
+    )
+    ld.add_action(set_container_name)
+
+    container = ComposableNodeContainer(
+        name=container_name,
+        namespace="",
+        package="rclcpp_components",
+        executable="component_container",
+        composable_node_descriptions=[],
+        condition=use_composability,
+        output="screen",
+    )
+    ld.add_action(container)
+
+    # ---------------
+    # Launch Includes
+    # ---------------
+
+    IncludeLaunchDescriptionWithCondition(ld, "art_perception_launch", "art_perception")
+    IncludeLaunchDescriptionWithCondition(ld, "art_localization_launch", "art_localization")
+    IncludeLaunchDescriptionWithCondition(ld, "art_planning_launch", "art_planning")
+    IncludeLaunchDescriptionWithCondition(ld, "art_control_launch", "art_control")
+    if(not IfCondition(PythonExpression([(GetLaunchArgument("use_sim"))]))):
```

Comment:
I added this if condition so that we don't get failed nodes when using sim. Only launch sim when use_sim is set, and only launch arduino/sensing nodes when use_sim is not set. This suppresses a lot of the image/... topics. I'm assuming those were being generated by the usb_cam, but not sure where they were coming from as the usb_cam seemed to be failing for me if an actual camera wasn't connected.